src/XR/features/WebXRLightEstimation.ts

```diff
@@ -0,0 +1,366 @@
+
+        if (this.directionalLight) {
+            this.directionalLight.dispose();
+            this.directionalLight = null;
+        }
+
+        if (this._reflectionCubeMap !== null) {
+            if (this._reflectionCubeMap._texture) {
+                this._reflectionCubeMap._texture.dispose();
+            }
+            this._reflectionCubeMap.dispose();
+            this._reflectionCubeMap = null;
+        }
+    }
+
+    protected _onXRFrame(_xrFrame: XRFrame): void {
+        if (this._xrLightProbe !== null) {
+            if (this.options.lightEstimationPollInterval) {
+                const now = Date.now();
+                if (now - this._lightEstimationPollTime < this.options.lightEstimationPollInterval) {
+                    return;
+                }
+                this._lightEstimationPollTime = now;
+            }
+            this._xrLightEstimate = _xrFrame.getLightEstimate(this._xrLightProbe);
+            if (this._xrLightEstimate) {
+                this._intensity = Math.max(1.0,
+                    this._xrLightEstimate.primaryLightIntensity.x,
+                    this._xrLightEstimate.primaryLightIntensity.y,
+                    this._xrLightEstimate.primaryLightIntensity.z);
+
+                const rhsFactor = this._xrSessionManager.scene.useRightHandedSystem ? 1.0 : -1.0;
+
+                // recreate the vector caches, so that the last one provided to the user will persist
+                if (this.options.disableVectorReuse) {
+                    this._lightDirection = new Vector3();
+                    this._lightColor = new Color3();
+                    if (this.directionalLight) {
+                        this.directionalLight.direction = this._lightDirection;
+                        this.directionalLight.diffuse = this._lightColor;
+                    }
+                }
+
+                this._lightDirection.copyFromFloats(
+                    this._xrLightEstimate.primaryLightDirection.x,
+                    this._xrLightEstimate.primaryLightDirection.y,
+                    this._xrLightEstimate.primaryLightDirection.z * rhsFactor
+                );
+                this._lightColor.copyFromFloats(
+                    this._xrLightEstimate.primaryLightIntensity.x / this._intensity,
+                    this._xrLightEstimate.primaryLightIntensity.y / this._intensity,
+                    this._xrLightEstimate.primaryLightIntensity.z / this._intensity
+                );
+
+                // direction from instead of direction to
+                this._lightDirection.negateInPlace();
+                // set the values after calculating them
+                if (this.directionalLight) {
+                    this.directionalLight.direction.copyFrom(this._lightDirection);
+                    this.directionalLight.intensity = Math.min(this._intensity, 1.0);
+                    this.directionalLight.diffuse.copyFrom(this._lightColor);

```

Comment:
I am wondering if they provide a unit or decay we should follow so we could force the correct light mode as default will work differently in standard and PBR ?

Reply:
no, they don't. they provide the following:

```
  readonly attribute Float32Array sphericalHarmonicsCoefficients;
  readonly attribute DOMPointReadOnly primaryLightDirection;
  readonly attribute DOMPointReadOnly primaryLightIntensity;
```
intensity represents RGB value.
The spherical harmonics is not being used here.

This light is optional. a dev could take the values and use them in a different way

Reply:
So lets force the mode to gltf which is probably the most used one :-) also sphericalHarmonicsCoefficients is AMAZING !!! as you could populate the ones in the environment texture with them !!!

Reply:
Done :-)